### ISSUE TITLE
Decoding `ConsumerGroupMemberMetadata` fails

### ISSUE THREAD

**Issue by random-dwi** [Shopify/sarama#1808]
##### Versions

| Sarama | Kafka | Go |
|--------|-------|----|
| v1.27.0 | 2.3.1 | 1.14.9 |

##### Problem Description

Decoding `ConsumerGroupMemberMetadata` fails with:
```
kafka: error decoding packet: invalid length
```

how to reproduce:
```
	var (
		err          error
		descriptions []*sarama.GroupDescription
	)

	if descriptions, err = admin.DescribeConsumerGroups(groupNames); err != nil {
		return nil, errors.Wrap(err, "failed to describe consumer groups")
	}

	for _, description := range descriptions {

		if description.ProtocolType != "consumer" {
			continue
		}

		for _, member := range description.Members {
			metaData, err := member.GetMemberMetadata() // here decoding fails
                }
```

I already tried to investigate what is going wrong. The [Protocol Guide](https://kafka.apache.org/protocol) only states:

| Field | Type | Description |
|----|----|----|
|member_metadata | BYTES | The metadata corresponding to the current group protocol in use.|

When the decoding of `ConsumerGroupMemberMetadata` happens all fields (`Version`, `Topics`, `UserData`) can be decoded but the buffer still contains unread bytes afterwards. Thats why the error is thrown.

I noticed that `Version` is `1`. Perhaps a schema with version `0` existed and with this the decoding worked. Sadly I cannot find any documentation of how the schema is defined.

I can help on this topic if someone can provide me a documentation of the schema used in `ConsumerGroupMemberMetadata`.

**Comment by dnwe** [comment]
@random-dwi I think this is a dupe of https://github.com/Shopify/sarama/issues/1717 and was a server-side broker issue rather than a client problem. It was a regression in Kafka 2.3.0, but was fixed in Kafka 2.4.0 under [KAFKA-9150](https://issues.apache.org/jira/browse/KAFKA-9150). I submitted https://github.com/apache/kafka/pull/8888 to have the fix backported to the 2.3 branch which was merged, but they haven't released another Kafka 2.3.x release since.

Note also that the Kafka Java adminclient only deserializes the member assignments, *not* the member metadata.
https://github.com/apache/kafka/blob/1d24e2e3b2a119997fc6805f9c754a5ee286d7b2/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L2960-L2964 so you probably don’t really want to either

**Comment by random-dwi** [comment]
@dnwe thank you for the quick response. I'm currently using this to get a list of all consumer groups that consume from a given topic. But instead of reading the topics from the member metadata I can read them from the member assignments I guess.
So I will close this one.
